### ISSUE TITLE
How to install hooks for submodule?

### ISSUE THREAD

**Issue by guglie** [pre-commit/pre-commit#2753]
I need to install the same hooks for parent repo and submodule, is this supported by pre-commit?
Sorry but I cant find the answer anywhere. Thanks

**Comment by asottile** [comment]
have you read the docs for install? and/or tried `--help` -- what have you tried so far (I can't help you if I don't know where you're confused)

**Comment by guglie** [comment]
Hello, thanks for the quick reply.
I read `pre-commit --help` and `pre-commit install --help` and [pre-commit.com](https://pre-commit.com)
But I don't get how to install pre-commit hooks for a submodule repository (if it's a supported feature).

Note: I want to install hooks inside `.git/modules/my-submodule/hooks/`

**Comment by asottile** [comment]
what have you tried that isn't working?

**Comment by guglie** [comment]
Didn't want to test randomly without reading the "correct/suggested" way somewhere so I only tried `pre-commit install` from the main repo which works only for the main repo.
I took your last message as an invitation to try more so feeling a bit more comfortable I did run
```
cd my-submodule
pre-commit install
```
which worked perfectly.

Hope this will be useful for other people getting stuck on this. 😊

**Comment by AlexanderSerov** [comment]
So answer to the original question is traverse submodules manually and execute `pre-commit install` for each? Is that right? We have a main repo with a lot of submodules. We have instructions for user to clone repo and execute `pre-commit install` once on main repo. So should we instruct user to execute pre-commit install on each submodule manually by `cd` to subfolder or there is a automated way to add hooks to all submodules at once after main repo cloned?

**Comment by asottile** [comment]
yes if you plan to do development in the submodules they are separate repos.  you can probably use `git submodule foreach` or `xargs`

**Comment by triplef** [comment]
I found this thread because I was also wondering about pre-commit’s handling of submodules.

In summary it sounds like with submodules you want to run something like this:

```shell
# install on parent repository
pre-commit install

# install on submodules, but don't complain if a submodule doesn't have a .pre-commit-config.yaml
git submodule foreach pre-commit install --allow-missing-config
```
